src/contentScripts/index.ts

```diff
@@ -1,8 +1,11 @@
 /* eslint-disable no-console */
 import { onMessage } from 'webext-bridge'
+import browser from 'webextension-polyfill'
 import { createApp } from 'vue'
 import App from './Content.vue'
 
+const isDev = browser.runtime.getManifest()?.content_security_policy?.includes('http://localhost') === true
```

Comment:
So tricky, why not try to [define](https://vitejs.dev/config/#define) a `__DEV__` constant? and it's even tree-shakable :smile:

Reply:
updated. thanks, I learned a lot from your review.